### PR TITLE
`Development`: Speed up client tests for the online code editor

### PR DESCRIPTION
**Pull request by MDK5**
#### General

- [x] Test run locally.

### Motivation, Context and Decription
Removes unneeded module imports from the code-editor-file-browser.component.spec.ts and converts sinon/chai to jest. I also removed imports from the other code-editor files where possible, but did not convert them.

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage
Should be unchanged

### CODE REVIEW

File: src/test/javascript/spec/component/code-editor/code-editor-actions.component.spec.ts [hunk 1]
```diff
@@ -35,9 +32,9 @@ describe('CodeEditorActionsComponent', () => {
     let updateFilesStub: jest.SpyInstance;
     let commitStub: jest.SpyInstance;
 
-    beforeEach(async () => {
+    beforeEach(() => {
         return TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, AceEditorModule, FeatureToggleModule],
+            imports: [ArtemisTestModule, AceEditorModule, FeatureToggleModule],
```
Comment (imstreetspirit) [inline src/test/javascript/spec/component/code-editor/code-editor-actions.component.spec.ts:37]: not sure if it fits for this test suite, maybe FeatureToggleModule can be removed here.. you can mock FeatureToggleDirective... In addition, you could also re-organize the assertions in this file. Some still uses toBeTruthy or toBeFalsy which are not recommended in guideline.

**Review by imstreetspirit — COMMENTED** [review]
I left inline comment, besides looks good 💪

File: src/test/javascript/spec/component/code-editor/code-editor-actions.component.spec.ts (hunk 1 above)
Comment (MDK5) [reply src/test/javascript/spec/component/code-editor/code-editor-actions.component.spec.ts]: I just checked, the directive itself cant be mocked, but the module can be replaced with only the directive

**Comment by imstreetspirit** [comment]
only very small style issues left, then I'll approve

File: src/test/javascript/spec/component/code-editor/code-editor-actions.component.spec.ts [hunk 2]
```diff
@@ -23,8 +17,11 @@ import { MockCodeEditorRepositoryFileService } from '../../helpers/mocks/service
 import { MockCodeEditorRepositoryService } from '../../helpers/mocks/service/mock-code-editor-repository.service';
 import { MockCookieService } from '../../helpers/mocks/service/mock-cookie.service';
 import { CommitState, EditorState } from 'app/exercises/programming/shared/code-editor/model/code-editor.model';
-import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
-import { MockPipe } from 'ng-mocks';
+import { MockModule} from 'ng-mocks';
```
Comment (MDK5) [inline src/test/javascript/spec/component/code-editor/code-editor-actions.component.spec.ts:20]:
```suggestion
import { MockModule } from 'ng-mocks';
```

File: src/test/javascript/spec/component/code-editor/code-editor-status.component.spec.ts [hunk 3]
```diff
@@ -1,40 +1,37 @@
 import { DebugElement } from '@angular/core/';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (MDK5) [inline src/test/javascript/spec/component/code-editor/code-editor-status.component.spec.ts:2]:
```suggestion
import { ComponentFixture, TestBed } from '@angular/core/testing';
```

**Review by imstreetspirit — APPROVED** [review]
looks good now 👍 🥳

**Review by DanielKainz — APPROVED** [review]
Code changes look good. Great Work :-)

**Pull request merged**